openc3/lib/openc3/accessors/binary_accessor.rb

```diff
@@ -115,6 +115,149 @@ def self.raise_buffer_error(read_write, buffer, data_type, given_bit_offset, giv
+          else
+            # Length = 3, value up to 62-bits
+            new_bit_size = 62
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 3)
+          end
+        else
+          if value <= 31 and value >= -32
+            # Length = 0, value up to 6-bits
+            new_bit_size = 6
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 0)
+          elsif value <= 8191 and value >= -8192
+            # Length = 1, value up to 14-bits
+            new_bit_size = 14
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 1)
+          elsif value <= 536870911 and value >= -536870912
+            # Length = 2, value up to 30-bits
+            new_bit_size = 30
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 2)
+          else
+            # Length = 3, value up to 62-bits
+            new_bit_size = 62
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 3)
+          end
+        end
+
+        # Later items need their bit_offset adjusted by the change in this item
+        adjustment = new_bit_size - current_bit_size
+      elsif item.data_type == :FLOAT
+        raise "Variable bit size not currently supported for FLOAT data type"
+      else
+        # STRING, BLOCK, or array types
+
+        # Calculate current bit size so we can preserve bytes after the item
+        length_item_value = @packet.read(item.variable_bit_size['length_item_name'], :CONVERTED)
+        current_bit_size = (length_item_value * item.variable_bit_size['length_bits_per_count']) + item.variable_bit_size['length_value_bit_offset']
+
+        # Calculate bits after this item
+        bits_with_item = item.bit_offset + current_bit_size
+        bits_after_item = (buffer.length * 8) - bits_with_item
+        if item.original_array_size
+          item.array_size = -bits_after_item
+        else
+          item.bit_size = -bits_after_item
+        end
+
+        new_bit_size = value.length * 8
+        length_value = (new_bit_size - item.variable_bit_size['length_value_bit_offset']) / item.variable_bit_size['length_bits_per_count']
+        @packet.write(item.variable_bit_size['length_item_name'], length_value)
+
+        # Later items need their bit_offset adjusted by the change in this item
+        adjustment = new_bit_size - current_bit_size
+      end
+
+      # Recalculate bit offsets after this item
+      if adjustment != 0 and item.bit_offset >= 0
+        found = false
+        @packet.sorted_items.each do |sitem|
+          if sitem.data_type == :DERIVED or sitem.bit_offset < item.bit_offset
+            # Skip items before this item and derived items and items with negative bit offsets
+            next
+          end
+          if sitem != item
```

Comment:
The <=> operator in structure_item is based on bit_offset so you might be able to remove this check and just change the above check to `sitem.bit_offset <= item.bit_offset`